packs/BP/scripts/network.ts

```diff
@@ -89,123 +89,122 @@ export class MachineNetwork extends DestroyableObject {
    * automatically sets each generator's storage to the amount it sent that was not received.
    * returns automatically if the object is not valid.
    */
-  private *send(): Generator<void, void, void> {
+  private *send(): Generator<undefined, void, unknown> {
     if (!this.isValid) return;
 
-    interface Target {
-      block: Block;
-      amount: number;
-      definition: InternalRegisteredMachine;
-    }
-
-    const targets: Record<string, Target[]> = {};
-
-    while (this.sendQueue.length) {
-      const queuedSend = this.sendQueue.pop()!;
-      if (!(queuedSend.type in targets)) {
-        const targetsArr: Target[] = [];
-
-        for (const block of this.connections.machines) {
-          if (
-            !block.hasTag(
-              `fluffyalien_energisticscore:consumer.${queuedSend.type}`,
-            ) &&
-            !block.hasTag("fluffyalien_energisticscore:consumer._any")
-          ) {
-            yield;
-            continue;
-          }
-
-          const definition = machineRegistry[block.typeId] as
-            | InternalRegisteredMachine
-            | undefined;
-
-          if (!definition) {
-            logWarn(
-              `can't send '${queuedSend.type}' to machine '${block.typeId}': this block is configured as a machine but it couldn't be found in the machine registry`,
-            );
-            yield;
-            continue;
-          }
-
-          const amount = getMachineStorage(block, queuedSend.type);
-
-          if (amount >= definition.maxStorage) {
-            yield;
-            continue;
-          }
-
-          targetsArr.push({
-            block,
-            amount,
-            definition,
-          });
-
-          yield;
+    // Calculate the amount of each type that is avaliable to send around.
+    const distribution: Record<string, number> = {};
+
+    this.sendQueue.forEach(send => {
+      const newValue = distribution[send.type] ?? 0;
+      distribution[send.type] = newValue + send.amount;
+    });
+
+    // Clear queue for next time.
+    this.sendQueue.splice(0, this.sendQueue.length);
+    const typesToDistribute = Object.keys(distribution);
+
+    interface Consumer { normalPriority: Block[], lowPriority: Block[] };
```

Comment:
the name `Consumer` is misleading as this interface contains multiple consumers. maybe name it `ConsumerGroup`?